kedro/framework/cli/cli.py

```diff
@@ -98,12 +100,13 @@ def docs():
 
 def _init_plugins():
     group = ENTRY_POINT_GROUPS["init"]
-    for entry_point in pkg_resources.iter_entry_points(group=group):
+    for entry_point in importlib_metadata.entry_points().select(group=group):
```

Comment:
Would it make sense to refactor this function a bit so it uses `load_entry_points` in utils.py? At a glance they look very similar.

Reply:
The original test for `_init_plugins` is actually incorrectly testing the `load` exception, it should be testing the error during hook initialisation instead.
```
init_hook = entry_point.load() # previously the the error was triggered here
init_hook()  # this is what we want to test instead
```